### PR TITLE
[@types\braintree-web] Updated HostedFields to version 3.51.0

### PR DESCRIPTION
**Pull request by brianroadifer**
Updated interfaces with changes to braintree-web.
Added deprecation notices to functions and interfaces parts that are deprecated.
Added new function calls.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://braintree.github.io/braintree-web/current/HostedFields.html>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by jamesg1** [comment]
I have just started using braintree-web and I noticed paypal-checkout is the latest version of paypal and the existing paypal definitions are deprecated 

There was a ticket opened a while back for typescript paypal-checkout. https://github.com/DefinitelyTyped/DefinitelyTyped/issues/18028

Should we add some definitions for this too as part of the 3.51.0 upgrade?

**Comment by brianroadifer** [comment]
Yeah, I can go back and run through the changes for paypal-checkout.

**Comment by jamesg1** [comment]
Happy to help too.

**Comment by brianroadifer** [comment]
Yeah, I will add you to my fork of it. Though we might need a separate branch to develop on, seeing that this is still waiting for the pull request to be approved off of the master.

**Comment by jamesg1** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/35610 from @schmod includes types for paypal-checkout. I like how the types are split up too to match the braintree-web repo.

**Comment by schmod** [comment]
Oh, yeah... I guess I can dust off that branch.

**Comment by brianroadifer** [comment]
@jamesg1 you would want to wait for a new pull request by @schmod then since theirs is doing  much more than what mine is.

File: .github/CODEOWNERS
Comment (uniqueiniquity) [inline .github/CODEOWNERS]: Please don't update codeowners manually - we'll update it regularly based on the headers in the individual packages.

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]
I'm not sure I follow the conversation going on in the main PR thread - is this ready to merge once the codeowners changes are removed?

**Comment by brianroadifer** [comment]
This is ready to merge once I make the necessary changes.

File: .github/CODEOWNERS
Comment (brianroadifer) [reply .github/CODEOWNERS]: Pulled from the latest CODEOWNERS from UPSTREAM

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]
The CODEOWNERS file still differs from `DefinitelyTyped:master`.
Please ensure CODEOWNERS is not shown in "files changed".

**Comment by brianroadifer** [comment]
How would I go about on resetting the file without getting merge conflicts?

**Comment by brianroadifer** [comment]
Sorry, the close and comment button is right next to the comment button.

**Comment by sandersn** [comment]
@brianroadifer can you try

```
git revert dcdb3b537131cf4a6b340f4f72e0191f8dda0b62
git revert 8af3afe61dfb6120b62594606bedb2c5ced5a3f1
```

Those are the two commit SHAs that are labelled "Update CODEOWNERS" in the commit history of the PR.

**Comment by brianroadifer** [comment]
I have reverted those commits

**Comment by sandersn** [comment]
OK, it looks like Travis is back.

**Pull request closed without merging**
